### PR TITLE
Add Test for Batch Message ThreadPool

### PR DESCRIPTION
**Pull request by dasahcc**
Add Test for Batch Message ThreadPool. For easy testing, change the private type to default type.

### CODE REVIEW

File: helix-core/src/test/java/org/apache/helix/messaging/handling/TestResourceThreadpoolSize.java [hunk 1]
```diff
@@ -113,6 +113,35 @@ public void TestThreadPoolSizeConfig() {
     }
   }
 
+  @Test
+  public void testBatchMessageThreadPoolSize() throws InterruptedException {
```
Comment (kishoreg) [inline helix-core/src/test/java/org/apache/helix/messaging/handling/TestResourceThreadpoolSize.java]: we need a test with number of resources with group message enabled > threapool size. This is the scenario where current code deadlocks. This test case does not cover that scenario

Comment (dasahcc) [inline helix-core/src/test/java/org/apache/helix/messaging/handling/TestResourceThreadpoolSize.java]: Thanks for suggestion. Just made the change for this test with the senario.

Comment (kishoreg) [inline helix-core/src/test/java/org/apache/helix/messaging/handling/TestResourceThreadpoolSize.java]: Thanks, do you see that this test case fails without my change. Might have to run it multiple times or add a delay in batch message handler before it executes the sub-messages to reproduce it.

Comment (dasahcc) [inline helix-core/src/test/java/org/apache/helix/messaging/handling/TestResourceThreadpoolSize.java]: Good suggestion! I tested it manually add the delay in batch message handle, it passed! Unfortunately, we does not have proper way to have a test BatchMessageHandler. So I just add a delay for state transitions. Hope this help.

**Pull request merged**